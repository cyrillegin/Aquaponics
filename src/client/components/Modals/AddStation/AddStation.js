import React, { useState } from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

/* eslint-disable prefer-destructuring */
const AddStation = ({ className, close }) => {
  const [input, setInput] = useState({});
  const [testSuccessfull, setSuccess] = useState(false);

  const handleInputChange = event =>
    setInput({
      ...input,
      [event.currentTarget.name]: event.currentTarget.value,
    });
  const handleTest = () => {
    const params = {
      ...input,
    };

    if (!params.name) {
      return;
    }

    fetch('/api/station/test', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(params),
    })
      .then(res => res.json())
      .then(res => {
        if (res.message === 'success') {
          setSuccess(true);
        } else {
          setSuccess(false);
        }
      });
  };

  const handleAdd = () => {
    const params = {
      ...input,
    };

    fetch('/api/station/', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(params),
    })
      .then(res => res.json())
      .then(res => {});
  };

  const preventClose = event => {
    event.preventDefault();
    event.stopPropagation();
  };

  return (
    <div className={className} onClick={close}>
      <div className="modal" onClick={preventClose}>
        <div className="title">Add Station</div>
        <div className="body">
          <div className="column">
            <div className="group">
              Name:
              <input type="text" name="name" onChange={handleInputChange} />
            </div>

            <div className="group">
              IP Adress:
              <input type="text" name="ipaddress" onChange={handleInputChange} />
            </div>
          </div>
          <div className="column">
            <div className="group">
              Description:
              <input type="text" ame="description" />
            </div>
            <div className="group">
              <button type="button" onClick={handleTest}>
                Test
              </button>
              <span className={`${testSuccessfull ? 'green' : 'red'}`} />
              <br />
              {testSuccessfull && (
                <button type="button" onClick={handleAdd}>
                  Add
                </button>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

AddStation.propTypes = {
  className: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired,
};

const styledAddStation = styled(AddStation)`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.75);
  z-index: 2;

  .modal {
    width: 600px;
    height: 601px;
    background: white;
    margin: 5rem auto;
    border-radius: 0.5rem;
    padding: 1rem;
    box-sizing: border-box;

    .body {
      display: flex;

      .column {
        flex: 1;

        .group {
          span {
            width: 10px;
            height: 10px;
            border-radius: 10px;
            margin: 0 20px;
            display: inline-block;

            &.red {
              background: red;
            }

            &.green {
              background: green;
            }
          }
        }
      }
    }
  }
`;

export default styledAddStation;
