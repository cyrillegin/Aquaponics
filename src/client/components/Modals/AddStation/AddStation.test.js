import { mount } from 'enzyme';
import React from 'react';
import AddStation from './AddStation';

describe('AddStation', () => {
  it('should render a snap shot', () => {
    const wrapper = mount(<AddStation />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
