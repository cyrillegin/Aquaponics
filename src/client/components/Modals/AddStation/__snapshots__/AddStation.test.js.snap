// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStation should render a snap shot 1`] = `"<div class=\\"css-1v2qv0l\\"><div class=\\"modal\\"><div class=\\"title\\">Add Station</div><div class=\\"body\\"><div class=\\"column\\"><div class=\\"group\\">Name:<input type=\\"text\\" name=\\"name\\"></div><div class=\\"group\\">Adress:<input type=\\"text\\" name=\\"address\\"></div></div><div class=\\"column\\"><div class=\\"group\\">Description:<input type=\\"text\\" ame=\\"description\\"></div><div class=\\"group\\">Port:<input type=\\"text\\" name=\\"port\\"></div><div class=\\"group\\"><button type=\\"button\\">Test</button><span class=\\"red\\"></span><br></div></div></div></div></div>"`;
